### ISSUE TITLE
SMT-LIB quoted symbols

### ISSUE THREAD

**Issue by selig** [vprover/vampire#307]
Vampire doesn't handle SMT-LIB  quoted symbols. From the standard:

"A quoted symbol is any sequence of whitespace characters and printable characters that starts and ends with | and does not contain | or \ ."

and

"Following Common Lisp’s conventions, enclosing a simple symbol in vertical bars does not produce a new symbol. This means for instance that `abc` and `|abc|` are the same symbol."

Note that these symbols can be sort or function symbols. Currently, Vampire will treat `abc` and `|abc|` as different symbols. Example:

```
(set-logic UF)
(declare-sort |T| 0)
(declare-fun  |f| (T) T)
(declare-const |a| T)
(assert (= a (f a)))
(check-sat)
```

**Comment by quickbeam123** [comment]
I am vaguely aware of this. Ideally, I think, we would want to fix the lexer, such that the parser does not need to care...

**Comment by selig** [comment]
Fixed by #308
